contracts/MarginBase.sol

```diff
@@ -312,7 +312,17 @@ contract MarginBase is MinimalProxyable, OpsReady, IMarginBaseTypes {
         }
     }
 
-    // @TODO https://github.com/Kwenta/margin-manager/issues/12
+    /// @notice accept a deposit amount and open new
+    /// futures market position(s) all in a single tx
+    /// @param _amount: amount of marginAsset to deposit into marginBase account
+    /// @param _newPositions: an array of UpdateMarketPositionSpec's used to modify active market positions
+    function depositAndDistribute(
+        uint256 _amount,
+        UpdateMarketPositionSpec[] memory _newPositions
+    ) external onlyOwner {
+        deposit(_amount);
```

Comment:
can you double check this? deposit has an onlyOwner modifier, in deposit() ,is `msg.sender` == account owner or does it equal this contract

Reply:
msg.sender is still EOA. I will add another test exhibiting that behavior though just to be clear

Reply:
Added test!